cmd/zoekt-sourcegraph-indexserver/sg.go

```diff
@@ -128,24 +168,53 @@ type sourcegraphClient struct {
 	// zero a value of 10000 is used.
 	BatchSize int
 
-	// Client is used to make requests to the Sourcegraph instance. Prefer to
+	// restClient is used to make requests to the Sourcegraph instance. Prefer to
 	// use .doRequest() to ensure the appropriate headers are set.
-	Client *retryablehttp.Client
+	restClient *retryablehttp.Client
+
+	// grpcClient is used to make requests to the Sourcegraph instance if gRPC is enabled.
+	grpcClient proto.IndexedSearchConfigurationServiceClient
 
 	// configFingerprint is the last config fingerprint returned from
 	// Sourcegraph. It can be used for future calls to the configuration
 	// endpoint.
 	configFingerprint string
 
+	// configFingerprintProto is the last config fingerprint (as GRPC) returned from
+	// Sourcegraph. It can be used for future calls to the configuration
+	// endpoint.
+	configFingerprintProto *proto.Fingerprint
+
 	// configFingerprintReset tracks when we should zero out the
 	// configFingerprint. We want to periodically do this just in case our
 	// configFingerprint logic is faulty. When it is cleared out, we fallback to
 	// calculating everything.
 	configFingerprintReset time.Time
+
+	// shouldUseGRPCFunc is a function which returns true if we should communicate with
+	// Sourcegraph via gRPC.
+	shouldUseGRPCFunc func() bool
```

Comment:
The idea behind making this a function was to make it possible to have dynamic configuration (example: control gRPC via the presence/absence of a specific text file) instead of having to change an environment variable / re-deploy the pod entirely.

Reply:
We've changed this option to be a fixed value on startup.